### DISCUSSION TITLE
Erreur de compilation

### DISCUSSION THREAD

**Discussion by bbjodel** [mathieucarbou/MycilaJSY#12] [General]
Bonjour Mathieu, j'ai essayé ce code pour faire fonctionner le JSY-MK-227 mais il y a une erreur a la compilation sur une librarie, voici l'erreur:
In file included from c:\users\ingrassia alfred\appdata\local\arduino15\packages\esp32\tools\xtensa-esp32-elf-gcc\1.22.0-97-gc752ad5-5.2.0\xtensa-esp32-elf\include\c++\5.2.0\shared_mutex:35:0,
                 from C:\Program Files (x86)\arduino-1.8.19\libraries\MycilaJSY\src/MycilaJSY.h:10,
                 from C:\Users\Alfred\Documents\Arduino 1.8.19\ESP32_TEST_JSY_MK_227\ESP32_TEST_JSY_MK_227.ino:3:
c:\users\alfred\appdata\local\arduino15\packages\esp32\tools\xtensa-esp32-elf-gcc\1.22.0-97-gc752ad5-5.2.0\xtensa-esp32-elf\include\c++\5.2.0\bits\c++14_warning.h:32:2: error: #error This file requires compiler and library support for the forthcoming ISO C++ 2014 standard. This support is currently experimental, and must be enabled with the -std=c++1y or -std=gnu++1y compiler options.
 #error This file requires compiler and library support for the forthcoming \
  ^
exit status 1
Erreur de compilation pour la carte DOIT ESP32 DEVKIT V1

D'autre part est ce que ce code est valable ?
Comment connecter le JSY-MK-1031 avec le JSY-MK-227 ?

#include <Arduino.h>
#include <ArduinoJson.h>
#include <MycilaJSY.h>

#define RX2 16
#define TX2 17

Mycila::JSY jsy;

void setup() {
Serial.begin(115200):

jsy.begin(Serial2, RX2, TX2);
}

void loop() {

if(jsy.read()) {
float v = jsy.data.single().voltage;
float i = jsy.data.single().current;
}

Serial.print(v, 1);
Serial.print(" - ");
Serial.println(i, 1);
}

Amicalement, Fred

**Comment by mathieucarbou** [comment]
Bonjour,

oui, voir dans le fichier PIO: il faut utiliser c++ 17 ou plus.

```ini
; Arduino 3.0.x
; platform = https://github.com/pioarduino/platform-espressif32/releases/download/51.03.05/platform-espressif32.zip
; Arduino 3.1.x
platform = https://github.com/pioarduino/platform-espressif32/releases/download/53.03.10-rc3/platform-espressif32.zip

build_flags = 
  -std=c++17
  -std=gnu++17
  -Wall -Wextra

build_unflags =
    -std=gnu++11
```

Note: C++ 17 de mémoire est d"office avec Arduino 3.. S'assurer d'utilier Arduino 3.

Note: PlatformIO n'est pas compatible Arduino 3 également.

**Reply by bbjodel** [reply]
J'ai téléchargé platform-espressif32-53.03.10-rc3 ou doit etre copié ce dossier ? C'est quoi le fichier PIO ou on le trouve ? excuse moi je suis un novice....Arduino 3 n'existe pas a ma connaissance il en version 2 je crois

**Reply by mathieucarbou** [reply]
Je suis désolé, je veux bien faire du support sur les librairies que je développe mais je n'ai absolument pas de temps pour ce genre de support.  Il y a de nombreux tutos en ligne qui montrent comment mettre en place un projet platformio avec vscode, ou Arduino IDE, etc. 
À vous de vous former sur le setup d'un projet Arduino en premier lieu :-)
